packages/features/bookings/Booker/components/BookEventForm/BookFormAsModal.tsx

```diff
@@ -9,11 +10,33 @@ import { useTimePreferences } from "../../../lib";
 import { useBookerStore } from "../../store";
 import { useEvent } from "../../utils/event";
 
-const BookEventFormWrapper = ({ children }: { onCancel: () => void; children: ReactNode }) => {
+const BookEventFormWrapper = ({ children, onCancel }: { onCancel: () => void; children: ReactNode }) => {
+  const { data } = useEvent();
+
+  return <BookEventFormWrapperComponent child={children} eventLength={data?.length} onCancel={onCancel} />;
+};
+
+const PlatformBookEventFormWrapper = ({
+  children,
+  onCancel,
+}: {
+  onCancel: () => void;
+  children: ReactNode;
+}) => {
+  return <BookEventFormWrapperComponent child={children} onCancel={onCancel} />;
+};
+
+export const BookEventFormWrapperComponent = ({
+  child,
+  eventLength,
+}: {
+  onCancel: () => void;
+  child: ReactNode;
+  eventLength?: number;
+}) => {
```

Comment:
event could be passed as a props here tbh, we already do it multiple times in the rest of the booker

Reply:
this would remove the need to split into platform / web wrapper

Reply:
We only need the length of the event here. The trpc handler that we have inside of the `useEvent` hook returns the event length property, however we don't return event length property in platform endpoint for event types. As ai result had to split this into platform / web wrapper